### PR TITLE
fix: Position of the Showcase.withWidget when it is near to bottom

### PR DESCRIPTION
**Pull request by kashifalaliwala-simformsolutions**

### CODE REVIEW

**Comment by BirjuVachhani** [comment]
@kashifalaliwala-simformsolutions Does this fix any open issues?

**Comment by kashifalaliwala-simformsolutions** [comment]
> @kashifalaliwala-simformsolutions Does this fix any open issues?

Yes, Position of the Showcase.withWidget is different on iPhone 11 #57
https://github.com/SimformSolutionsPvtLtd/flutter_showcaseview/issues/57

**Comment by BirjuVachhani** [comment]
@kashifalaliwala-simformsolutions Great, I am linking that issue with this PR.

File: lib/tooltip_widget.dart [hunk 1]
```diff
@@ -35,12 +36,19 @@ class ToolTipWidget extends StatelessWidget {
...
+class _ToolTipWidgetState extends State<ToolTipWidget> {
+  Offset position;
+
   bool isCloseToTopOrBottom(Offset position) {
     double height = 120;
-    if (contentHeight != null) {
-      height = contentHeight;
+    if (widget.contentHeight != null) {
+      height = widget.contentHeight;
     }
```
Comment (BirjuVachhani) [inline lib/tooltip_widget.dart:50]: This could be combined with the use of `??` operator for more readability.

File: lib/tooltip_widget.dart [hunk 2]
```diff
@@ -52,8 +60,8 @@ class ToolTipWidget extends StatelessWidget {
   }
 
   double _getTooltipWidth() {
-    double titleLength = title == null ? 0 : (title.length * 10.0);
-    double descriptionLength = (description.length * 7.0);
+    double titleLength = widget.title == null ? 0 : (widget.title.length * 10.0);
+    double descriptionLength = (widget.description.length * 7.0);
```
Comment (BirjuVachhani) [inline lib/tooltip_widget.dart:64]: Brackets `()` are redundant here.

File: lib/tooltip_widget.dart [hunk 3]
```diff
@@ -62,80 +70,85 @@ class ToolTipWidget extends StatelessWidget {
...
+    super.initState();
+    position = widget.offset;
+  }
   @override
   Widget build(BuildContext context) {
-    final contentOrientation = findPositionForContent(offset);
+    final contentOrientation = findPositionForContent(position);
     final contentOffsetMultiplier = contentOrientation == "BELOW" ? 1.0 : -1.0;
-    isArrowUp = contentOffsetMultiplier == 1.0 ? true : false;
+    ToolTipWidget.isArrowUp = contentOffsetMultiplier == 1.0 ? true : false;
```
Comment (BirjuVachhani) [inline lib/tooltip_widget.dart:132]: Redundant turnary`(?:)` operator here.

File: lib/tooltip_widget.dart [hunk 4]
```diff
@@ -236,28 +257,28 @@ class ToolTipWidget extends StatelessWidget {
...
-          ).animate(animationOffset),
-          child: isArrowUp
+          ).animate(widget.animationOffset),
+          child: ToolTipWidget.isArrowUp
               ? Icon(
-                  Icons.arrow_drop_up,
-                  color: tooltipColor,
-                  size: 50,
-                )
+            Icons.arrow_drop_up,
```
Comment (BirjuVachhani) [inline lib/tooltip_widget.dart:271]: should apply condition to the icon property only. Thus we can avoid 3 duplicated lines.

**Comment by vipulasri** [comment]
Any update on this issue?

File: lib/tooltip_widget.dart (hunk 1 above)
Comment (kashifalaliwala-simformsolutions) [reply lib/tooltip_widget.dart]: Added ?? operator

File: lib/tooltip_widget.dart (hunk 2 above)
Comment (kashifalaliwala-simformsolutions) [reply lib/tooltip_widget.dart]: Removed redundant brackets

File: lib/tooltip_widget.dart (hunk 3 above)
Comment (kashifalaliwala-simformsolutions) [reply lib/tooltip_widget.dart]: Removed ternary operator

File: lib/tooltip_widget.dart (hunk 4 above)
Comment (kashifalaliwala-simformsolutions) [reply lib/tooltip_widget.dart]: Added check inside icon property and altered icons

**Comment by BirjuVachhani** [comment]
@kashifalaliwala-simformsolutions Thanks for your contribution. Keep up the good work.

**Comment by BirjuVachhani** [comment]
@vipulasri This will be available with #63.

**Comment by vipulasri** [comment]
@BirjuVachhani thanks for the intimation.

**Pull request merged**